specification/src/main/asciidoc/webprofile/jpa-cdi/CDI-JPA.adoc

```diff
@@ -0,0 +1,148 @@
+- no interceptor bindings,
+- a bean implementation which satisfies the requirements of
+this specification for a container-managed entity manager
+factory.
+
+Furthermore, the container must make available five beans with:
+
+- bean types _CriteriaBuilder_, _PersistenceUnitUtil_, _Cache_,
+_SchemaManager_, and _Metamodel_, respectively,
+- the qualifiers specified by _qualifier_ XML elements in
+_persistence.xml_, or _jakarta.enterprise.inject.Default_,
+if no qualifiers are explicitly specified,
+- scope _jakarta.enterprise.context.Dependent_,
+- no interceptor bindings,
+- a bean implementation which simply obtains the instance of
+the bean type by calling the appropriate getter method of
+the _EntityManagerFactory_ bean.
+
+==== Obtaining an Entity Manager Factory using CDI
+
+A container environment might feature built-in integration of
+Jakarta Persistence with the CDI bean manager, allowing
+injection of a container-managed entity manager factory using
+the annotation _jakarta.inject.Inject_. In the Jakarta EE
+environment, the container must feature built-integration of
+Jakarta Persistence with the CDI bean manager.
+
+For each persistence unit, a container which features such
+built-in integration must make available a bean with:
+
+- bean type _EntityManagerFactory_,
+- the qualifiers specified by _qualifier_ XML elements in
+_persistence.xml_, or _jakarta.enterprise.inject.Default_,
+if no qualifiers are explicitly specified,
+- scope _jakarta.enterprise.context.ApplicationScoped_,
+- bean name given by the name of the persistence unit,
+- no interceptor bindings,
+- a bean implementation which satisfies the requirements of
+this specification for a container-managed entity manager
+factory.
+
+Furthermore, the container must make available five beans with:
+
+- bean types _CriteriaBuilder_, _PersistenceUnitUtil_, _Cache_,
+_SchemaManager_, and _Metamodel_, respectively,
+- the qualifiers specified by _qualifier_ XML elements in
+_persistence.xml_, or _jakarta.enterprise.inject.Default_,
+if no qualifiers are explicitly specified,
+- scope _jakarta.enterprise.context.Dependent_,
+- no interceptor bindings,
+- a bean implementation which simply obtains the instance of
+the bean type by calling the appropriate getter method of
+the _EntityManagerFactory_ bean.
+
+==== persistence.xml Extended Schema
+This integration section defines an extension of the Persistence specification schema to add scope and qualifier elements.
+
+[source,xml]
+----
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- persistence.xml schema -->
+<xs:schema targetNamespace="https://jakarta.ee/xml/ns/persistence"
+            xmlns:xs="http://www.w3.org/2001/XMLSchema"
+            xmlns:persistence="https://jakarta.ee/xml/ns/persistence"
+            elementFormDefault="qualified"
+            attributeFormDefault="unqualified"
+            version="3.0">
+
+    <xs:include schemaLocation="persistence_3_0.xsd" />
```

Comment:
I am not a big fan of this way of extending schema - the drawback I see here is that the schema has to be updated every time the original one changes.

I'm thinking about an alternative - what if we add an extension point to the persistence schema itself somewhere? Sth like:

```
    <xsd:element name="persistence">
        <xsd:complexType>
            <xsd:sequence>
                <xsd:element name="persistence-unit" minOccurs="1" maxOccurs="unbounded">
                    <xsd:complexType>
                        <xsd:sequence>
                            <xsd:element name="description" type="xsd:string" minOccurs="0"></xsd:element>
                            <xsd:element name="provider" type="xsd:string" minOccurs="0"></xsd:element>
                            ...
                            <xsd:element name="properties" minOccurs="0">
                           <xsd:any namespace="##other" processContents="lax"/>
                        </xsd:sequence>
                    </xsd:complexType>
                </xsd:element>
            </xsd:sequence>
        </xsd:complexType>
    </xsd:element name="persistence">
...
```
the usage then could look like:
```
    <persistence-unit xsi:type="persistenceUnitExtType" name="my-unit" xmlns:cdi="https://jakarta.ee/xml/ns/persistence-cdi">
...
        <cdi:scope>com.example.jpa.ACustomScope</cdi:scope>
        <cdi:qualifier>com.example.jpa.CustomQualifier</cdi:qualifier>
    </persistence-unit>
</persistence>
```

The question is - does this _extension_ need to be available through `PersistenceUnitInfo`? If so, how? Ideally, it should be `org.w3c.dom.Node`, but I don't think I like the use of that type here

Reply:
If it does not need to be available via PersistenceUnitInfo, then there does not need for an extension to persistence.xml. In that case we could just require the info be passed into the Map passed to the factory method. Otherwise, your suggestion about the alternative syntax looks good to me.

Reply:
```xml
  <persistence-unit name="my-unit" xmlns:cdi="https://jakarta.ee/xml/ns/persistence-cdi">
     ...
    <cdi:scope>com.example.jpa.ACustomScope</cdi:scope>
    <cdi:qualifier>com.example.jpa.CustomQualifier</cdi:qualifier>
  </persistence-unit>
````

This looks nice indeed.